src/org/labkey/test/components/ui/EntityBulkInsertDialog.java

```diff
@@ -88,6 +91,43 @@ public String getFieldWithId(String id)
         return getWrapper().getFormElement(Locator.tagWithId("input", id));
     }
 
+    public EntityBulkInsertDialog setDateField(String fieldKey, String dateString)
+    {
+        Input input = elementCache().textInput(fieldKey);
+        input.set(dateString);
+        input.getComponentElement().sendKeys(Keys.ENTER);               // enter to collapse the datepicker
+        getWrapper().waitFor(()-> !isDatePickerExpanded(), 1500);
+        return this;
+    }
+
+    private boolean isDatePickerExpanded()
+    {
+        return Locator.tagWithClass("div", "react-datepicker-popper").existsIn(this);
+    }
+
+    public String getDateField(String fieldKey)
+    {
+        return elementCache().textInput(fieldKey).get();
+    }
+
+    public EntityBulkInsertDialog setBooleanField(String fieldKey, boolean checked)
+    {
+        Checkbox box = getCheckBox(fieldKey);
+        box.set(checked);
+        return this;
+    }
+
+    public boolean getBooleanField(String fieldKey)
+    {
+        return getCheckBox(fieldKey).get();
+    }
+
+    private Checkbox getCheckBox(String fieldKey)
+    {
+        WebElement row = elementCache().formRow(fieldKey);
+        return new Checkbox(elementCache().checkBoxLoc.findElement(row));
+    }
+
```

Comment:
Add `setCheckbox`?

Reply:
It's already handled in `setBooleanField , getBooleanField`